From d1054de2d5c21cf109b445a3af5cb1bf99d98f21 Mon Sep 17 00:00:00 2001
From: ChiaYen Kan <kcygamil@gmail.com>
Date: Mon, 6 Mar 2023 14:37:55 +0800
Subject: [PATCH] fix link error in msys2

---
 configure.ac | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/configure.ac b/configure.ac
index 10acaba0..1e16dd92 100644
--- a/configure.ac
+++ b/configure.ac
@@ -208,6 +208,15 @@ if test "x$have_cppunit" != "xyes"; then
   AC_MSG_WARN([$CPPUNIT_PKG_ERRORS])
 fi
 
+# Check iconv
+AC_CHECK_LIB([iconv], [iconv_open], , [ac_found_iconf=no])
+if test "x$ac_found_iconf" = "xno"; then
+  AC_CHECK_LIB([iconv], [libiconv_open], , [ac_found_iconf=no])
+fi
+if test "x$ac_found_iconf" = "xyes"; then
+  LIBS="-liconv $LIBS"
+fi
+
 # Check availability of libz
 if test "x$with_libz" = "xyes"; then
   PKG_CHECK_MODULES([ZLIB], [zlib >= 1.2.3], [have_zlib=yes], [have_zlib=no])
@@ -503,6 +512,13 @@ if test "x$with_libcares" = "xyes"; then
                     [have_libcares=no])
   if test "x$have_libcares" = "xyes"; then
     AC_DEFINE([HAVE_LIBCARES], [1], [Define to 1 if you have libcares.])
+    
+    case "$host" in
+      *mingw*)
+        LIBCARES_LIBS="$LIBCARES_LIBS -liphlpapi"
+      ;;
+    esac
+
     save_LIBS=$LIBS
     save_CPPFLAGS=$CPPFLAGS
     LIBS="$LIBCARES_LIBS $LIBS"
-- 
2.30.0.windows.2

